discovery_engine_core/providers/src/newscatcher/models.rs

```diff
@@ -0,0 +1,203 @@
+    /// Important: This parameter is not deducted on a per-article level:
+    /// it is deducted on the per-publisher level.
+    #[serde(default, deserialize_with = "deserialize_topic")]
+    pub topic: Topic,
+
+    /// The country of the publisher.
+    #[serde(default, deserialize_with = "deserialize_null_default")]
+    pub country: String,
+
+    /// The language of the article.
+    #[serde(default, deserialize_with = "deserialize_null_default")]
+    pub language: String,
+
+    /// While Newscatcher claims to have some sort of timezone support in their
+    /// [API][<https://docs.newscatcherapi.com/api-docs/endpoints/search-news>] (via the
+    /// `published_date_precision` attribute), in practice they do not seem to be supplying any
+    /// sort of timezone information. As a result, we provide NaiveDateTime for now.
+    #[serde(
+        default = "default_published_date",
+        deserialize_with = "deserialize_naive_date_time_from_str"
+    )]
+    pub published_date: NaiveDateTime,
+}
+
+fn default_published_date() -> NaiveDateTime {
+    chrono::naive::MIN_DATETIME
+}
+
+/// Null-value tolerant deserialization of `NaiveDateTime`
+fn deserialize_naive_date_time_from_str<'de, D>(deserializer: D) -> Result<NaiveDateTime, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let opt: Option<String> = Option::deserialize(deserializer)?;
+    opt.map_or_else(
+        || Ok(NaiveDateTime::from_timestamp(0, 0)),
+        |s| NaiveDateTime::parse_from_str(&s, "%Y-%m-%d %H:%M:%S").map_err(de::Error::custom),
+    )
+}
+
+/// Null-value tolerant deserialization of rank
+fn deserialize_rank<'de, D>(deserializer: D) -> Result<u64, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let opt = Option::deserialize(deserializer)?;
+    Ok(opt.unwrap_or(u64::MAX))
+}
+
+/// Null-value tolerant deserialization of topic
+fn deserialize_topic<'de, D>(deserializer: D) -> Result<Topic, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let opt = Option::deserialize(deserializer)?;
+    Ok(opt.unwrap_or(Topic::News))
+}
+
+impl From<Article> for crate::Article {
+    fn from(source: Article) -> Self {
+        crate::Article {
+            title: source.title,
+            snippet: source.excerpt,
+            url: source.link,
+            source_domain: source.source_domain,
+            date_published: source.published_date,
+            image: source.media,
+            rank: source.rank,
+            score: source.score,
+            country: source.country,
+            language: source.language,
+            topic: source.topic.to_string(),
+        }
+    }
+}
+
+impl crate::Article {
+    /// Loads newscatcher articles from JSON and returns them in `Article` representation
+    pub fn load_from_newscatcher_json_representation(json: &str) -> Result<Vec<Self>, Error> {
+        let articles: Vec<Article> = serde_json::from_str(json).map_err(Error::Decoding)?;
+
+        Ok(articles.into_iter().map(Into::into).collect())
```

Comment:
:art: the result could be mapped directly (`map_into()` requires `Itertools`)
```suggestion
        serde_json::from_str::<Vec<Article>>(json)
            .map(|articles| articles.into_iter().map_into().collect())
            .map_err(Error::Decoding)
```